httphandlers/http_handlers_test.go

```diff
@@ -96,18 +98,75 @@ func TestWriteLog(t *testing.T) {
-				"X-Request-Id": {"KnownTransactionId"},
-				"x-api-key":    {"ignore-key"},
-				"data-header":  {"test-header"},
+				"Referer":                     {"http://example.com"},
+				"User-Agent":                  {"User agent"},
+				"X-Request-Id":                {"KnownTransactionId"},
+				"x-api-key":                   {"ignore-key"},
+				"data-header":                 {"test-header"},
+				"Accept":                      {"*/*"},
+				"Accept-Encoding":             {"gzip"},
+				"Cache-Control":               {"no-cache"},
+				"Cdn-Loop":                    {"Fastly"},
+				"Connection":                  {"close"},
+				"Content-Length":              {"73"},
+				"Content-Type":                {"application/json"},
+				"Fastly-Client-Ip":            {"1.1.1.1"},
+				"Fastly-Ff":                   {"something to ignore"},
+				"Fastly-Orig-Accept-Encoding": {"gzip", "deflate"},
+				"Fastly-Ssl":                  {"1"},
+				"Fastly-Temp-Xff":             {"1.1.1.1", "2.2.2.2"},
+				"Postman-Token":               {"816da60e-d4be-47b8-ad49-387648169cec"},
+				"X-Forwarded-For":             {"1.1.1.1", "2.2.2.2"},
+				"X-Forwarded-Host":            {"test.ft.com"},
+				"X-Forwarded-Port":            {"443"},
+				"X-Forwarded-Proto":           {"https"},
+				"X-Forwarded-Server":          {"cache-vie11111-VIE"},
+				"X-Original-Request-Url":      {"test.ft.com"},
+				"X-Timer":                     {"S1589964637.859311,VS0"},
+				"X-Varnish":                   {"58679414", "15107515", "1840312"},
+				"X-Varnishpassthrough":        {"true"},
+			},
+			expectedLog: `{"host":"192.168.100.11", "level":"info","method":"GET","protocol":"HTTP/1.1",
+				"referer":"http://example.com","size":100,"status":200,"transaction_id":"KnownTransactionId",
+				"uri":"/","userAgent":"User agent","service_name":"test-service", 
+				"headers":{ "Accept":"*/*", "Accept-Encoding":"gzip", "Cache-Control":"no-cache", 
+					"Content-Type":"application/json", "Data-Header":"test-header",
+					"Postman-Token":"816da60e-d4be-47b8-ad49-387648169cec",
+					"X-Forwarded-For":"1.1.1.1, 2.2.2.2", "X-Forwarded-Host":"test.ft.com", "X-Forwarded-Port":"443",
+					"X-Forwarded-Proto":"https", "X-Forwarded-Server":"cache-vie11111-VIE",
+					"X-Original-Request-Url":"test.ft.com", "X-Varnishpassthrough":"true"}}`,
+		},
+		{
+			name:       "standard case with custom list headers",
+			url:        "http://example.com",
+			respTime:   time.Millisecond * 123,
+			remoteAddr: "192.168.100.11",
+			headers: map[string][]string{
+				"Referer":        {"http://example.com"},
+				"User-Agent":     {"User agent"},
+				"X-Request-Id":   {"KnownTransactionId"},
+				"x-api-key":      {"ignore-key"},
+				"allowed-header": {"test-header"},
+				"denied-header":  {"ignore-key"},
+			},
+			deniedHeaders: func(key string) bool {
+				filter := map[string]bool{
+					"allowed-header": true,
+					"denied-header":  false,
+				}
+				for f, b := range filter {
```

Comment:
We are only iterating over `filter` so it might be more appropriate for it to be a slice instead of a map.

Reply:
The idea was to show that we can have both allow and deny lists. Depending on what the user wants

Reply:
Oops, my bad, we are not only iterating over `filter` but we are using the boolean values as well. Still it seems strange for the `deniedHeaders` function to have allow logic, if the user wants to allow something then it just should not be present in the `filter` collection and/or in the `deniedHeaders` function.